### DISCUSSION TITLE
M5Stack carputer - esp32_midi_sampler adaptation?

### DISCUSSION THREAD

**Discussion by ayskura** [marcel-licence/esp32_midi_sampler#100] [General]
I just ordered an M5Stack cardputer, just becasue looks awesome and car record audio. Would be possible to run es32_midi_sampler on it? what would I need to adapt? 

https://shop.m5stack.com/products/m5stack-cardputer-kit-w-m5stamps3

**Comment by marcel-licence** [comment]
I would expect this to work. My project is very flexible. I also started the multiplatform version (which includes the soundfont loader). Do you know if there is an audio codec? Is there some I2C setup required?

**Reply by ayskura** [reply]
I still don't have the unit, but these are the specs

M5StampS3 Microcontroller.
56 key keyboard.
1.14 Inch TFT Screen.
Cavity speaker and SPM1423 digital MEMS microphone 
Infrared emission tube is used for infrared control interaction.
HY2.0-4P port for connecting and expanding I2C sensors.
Micro SD card slot for expanding storage space.

lookin at the examples here 
https://github.com/m5stack/M5Cardputer/blob/master/examples/Basic/buzzer/buzzer.ino

seem it's possible to access directly the buzzer, which I would assume it could be wired as output somehow.

the microphone is integrated, the only thing missing is MIDI

**Reply by marcel-licence** [reply]
It is an interesting idea to use the buzzer for audio output. I am not sure how it would sound like when trying to use this for digital to analogue conversion. Usually you use a digital signal only to generate beep tones. Do you have a certain idea in mind how you would like to generate your audio signal? Would you like to generate a signal like in the demo videos?

**Reply by ayskura** [reply]
for what I can hear from the examples, the buzzer is able to reproduce the audio recorded by the microphone in quite a good way - I still don't receive the unit, but it seem like it would make a perfect portable music sampler :) I will update once I got the unit as this might be a nice adaptation to make

**Reply by marcel-licence** [reply]
Do you have a video link to an example? I tried to output a PWM signal to a piezzo. Seems it can produce a quiet sound.
If you have a project which pushes samples to an audio output buffer you could integrate the sampler.

**Reply by ayskura** [reply]
I received the cardputer and it's really awesome, I was able to run this example where it constanly samples then when you press the A button it plays back via built'in buzzer the audio, which is quite high quality. It's easy to plug the buzzer connectors to an audio output so that wouldn't be a problem. just I wonder how to adapt the midi in and the whole sampler project. It would be awesome if it could display the waveform too as the screen is really crisp even if small. 

the example i ran was this one https://github.com/m5stack/M5Cardputer/tree/master/examples/Basic/mic

**Reply by marcel-licence** [reply]
Hi, it would be hard for me to implement and test because I do not own this device.
What kind of display is used here? Do you use the TFT_eSPI lib or similar to drive the display?
